### DISCUSSION TITLE
Eglot cannot find definitions of multiple layers of identifiers in the Rust std library

### DISCUSSION THREAD

**Discussion by rui-r-duan** [joaotavora/eglot#1254] [Q&A] [answered]
From my Rust code, I can use (eglot + rust-analyzer) to jump to the definition of a std function by pressing "M-.", for example `next`. But in the std library source file opened, if I want to find the definition of another identifier by pressing "M-.", Emacs prompts to ask me to open an ETAG file. If I run "M-x eglot" in the std library source file, then it may work for some identifiers in the source file, but when it jumps to other std library source files, eglot cannot find the definitions of the identifiers in them again, Emacs asks for an ETAG file again.

I tested the same series of identifiers in VSCode with rust-analyzer, it worked much better than using eglot.

I'm wondering why. Is there any simple configuration that can make it on a par with VSCode for finding definitions?

My environment: Emacs 29.0.92 built in my GNU/Linux Trisquel 11.

**Comment by joaotavora** [comment, marked as answer]
The standard library source file isn't in your project, found by Xref and Eglot was pointed out by the server but lives outside your project.  As explained in the manual, such files are by definition not managed by Eglot.  Pressing `M-x eglot` there will start a new Eglot session, which is not what you want.  

You may want to set `eglot-extend-to-xref` to t.  This creates an exception to the above rule.  It has implications when more than one project is being used at once, but normally it solves this problem. (No idea how VSCode solves it)

**Reply by rui-r-duan** [reply]
Following your guide, I set `eglot-extend-to-xref` to `t`, then the result is exactly what I want. I'm satisfied with it. After the change, I  found the document of `eglot-extend-to-xref` in `info`.  Thank you so much!
